timbermill-java/timbermill-local/src/main/java/com/datorama/oss/timbermill/ElasticsearchClient.java

```diff
@@ -942,11 +938,170 @@ long countByName(String name, String env) throws IOException {
+            return true;
+        }
+
+        if (storeSize.length() < 2) {
+            LOG.info("Store size {} is not in the right format", storeSize);
+            return false;
+        }
+
+        String sizeStr = storeSize.substring(0, storeSize.length() - 2);
+        double size = Double.parseDouble(sizeStr);
+        return size < maxIndexSizeInGB;
+    }
+
+    private void moveIndexAnotherIndex(String sourceIndex, String destIndex) {
+        LOG.info("Moving index {} to {}", sourceIndex, destIndex);
+        ReindexRequest reindexRequest = new ReindexRequest();
+        reindexRequest.setConflicts("proceed");
+        reindexRequest.setSourceIndices(sourceIndex);
+        reindexRequest.setDestIndex(destIndex);
+        reindexRequest.setDestOpType("create");
+        reindexRequest.setSlices(10);
+        reindexRequest.setTimeout(TimeValue.timeValueMinutes(10));
+
+        BoolQueryBuilder boolQuery = QueryBuilders.boolQuery();
+        RangeQueryBuilder rangeQuery = QueryBuilders.rangeQuery(TTL_FIELD).gte("now");
+        boolQuery.must(rangeQuery);
+
+        reindexRequest.setSourceQuery(boolQuery);
+
+        TaskId taskId;
+
+        try {
+            TaskSubmissionResponse taskSubmissionResponse = client.submitReindexTask(reindexRequest, RequestOptions.DEFAULT);
+            taskId = new TaskId(taskSubmissionResponse.getTask());
+        } catch (Exception e) {
+            LOG.error("Failed to submit reindex task from index {} to index {}", sourceIndex, destIndex, e);
+            return;
+        }
+
+        try {
+            if (isReindexSuccess(taskId)) {
+                deleteIndex(sourceIndex);
+            }
+        } catch (Exception e) {
+            LOG.error("Failed to wait for reindex task from index {} to index {}", sourceIndex, destIndex, e);
+        }
+    }
+
+    private boolean isReindexSuccess(TaskId taskId) {
+        InputStream content = null;
+        try {
+            while (true) {
+                Request request = new Request("GET", "/_tasks/" + taskId.toString());
+                Response response = client.getLowLevelClient().performRequest(request);
+                content = response.getEntity().getContent();
+                String json = IOUtils.toString(content);
+                JsonObject jsonObject = new JsonParser().parse(json).getAsJsonObject();
+                boolean isCompleted = jsonObject.get("completed").getAsBoolean();
+                if (isCompleted) {
+                    JsonObject responseObject = jsonObject.get("response").getAsJsonObject();
+                    int created = responseObject.get("created").getAsInt();
+                    int numOfFailures = responseObject.get("failures").getAsJsonArray().size();
+                    LOG.info("Reindex task completed. Created {} documents. Num of failures {}", created, numOfFailures);
+                    return numOfFailures == 0;
+                } else {
+					Thread.sleep(1000);
```

Comment:
I think you should sleep for longer, let's say 10 secs

Reply:
correct my bad, fixed